cpp/src/io/parquet/experimental/hybrid_scan_impl.cpp

```diff
@@ -69,6 +97,51 @@ void hybrid_scan_reader_impl::setup_page_index(
   _metadata->setup_page_index(page_index_bytes);
 }
 
+void hybrid_scan_reader_impl::select_columns(read_mode read_mode,
+                                             parquet_reader_options const& options)
+{
+  // Strings may be returned as either string or categorical columns
+  auto const strings_to_categorical = options.is_enabled_convert_strings_to_categories();
+  auto const use_pandas_metadata    = options.is_enabled_use_pandas_metadata();
+  auto const timestamp_type_id      = options.get_timestamp_type().id();
+
+  // Select only columns required by the filter
+  if (read_mode == read_mode::FILTER_COLUMNS) {
+    if (_is_filter_columns_selected) { return; }
+    // list, struct, dictionary are not supported by AST filter yet.
+    _filter_columns_names =
+      cudf::io::parquet::detail::get_column_names_in_expression(options.get_filter(), {});
+    // Select only filter columns using the base `select_columns` method
+    std::tie(_input_columns, _output_buffers, _output_column_schemas) = _metadata->select_columns(
+      _filter_columns_names, {}, use_pandas_metadata, strings_to_categorical, timestamp_type_id);
+
+    _is_filter_columns_selected  = true;
+    _is_payload_columns_selected = false;
+  } else {
+    if (_is_payload_columns_selected) { return; }
+
+    std::tie(_input_columns, _output_buffers, _output_column_schemas) =
+      _metadata->select_payload_columns(options.get_columns(),
+                                        _filter_columns_names,
+                                        use_pandas_metadata,
+                                        strings_to_categorical,
+                                        timestamp_type_id);
+
+    _is_payload_columns_selected = true;
+    _is_filter_columns_selected  = false;
+  }
+
+  CUDF_EXPECTS(_input_columns.size() > 0 and _output_buffers.size() > 0, "No columns selected");
+
+  // Clear the output buffers templates
+  _output_buffers_template.clear();
+
+  // Save the states of the output buffers for reuse.
+  for (auto const& buff : _output_buffers) {
+    _output_buffers_template.emplace_back(inline_column_buffer::empty_like(buff));
+  }
```

Comment:
transform?

Reply:
Copy-pasted stuff. Let me change here and at source